### ISSUE TITLE
onClear not working in SearchField in react-area-components 

### ISSUE THREAD

**Issue by rhulse** [adobe/react-spectrum#6509]
### Provide a general summary of the issue here

onClear on SearchForm does not fire

### 🤔 Expected Behavior?

The onClear handler is not fired

### 😯 Current Behavior

When test is typed an x appears. Clicking on the x clears the text but onClear is not fired.

### 🔦 Context

After a search has been run, if the user click the x to clear the terms, the handler is needed to clear the search and also remove any filters.

### 🖥️ Steps to Reproduce

Attach a function to the onClear handler. It is never called.

### Version

    "react-aria-components": "^1.2.1",

### What browsers are you seeing the problem on?

Other

### If other, please specify.

Arc

### What operating system are you using?

MacOS

**Comment by snowystinger** [comment]
I'm unable to reproduce, onClear is being called for me https://codesandbox.io/p/sandbox/xenodochial-parm-mj4pfr?file=%2Fsrc%2FApp.js%3A7%2C56

**Comment by rhulse** [comment]
Thanks for looking into this.

I have stripped back the code in my app to just what you have in that example, and what I am seeing is an extra X appearing in the input field. Because I started with minimal code I just assumed that was the x the triggered the event. 

A click of the x in the <Button> does return the event as expected, but it is a mystery where the other x is coming from.

https://github.com/adobe/react-spectrum/assets/30732/84572604-9349-4a8a-a649-97c07d0b3b42

Any guidance would be appreciated.
